presto-tests/src/main/java/com/facebook/presto/tests/ResultWithQueryId.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+public class ResultWithQueryId<T>
```

Comment:
Don't think so, as it's used as the output of H2QueryRunner and in AbstractTestHiveClient#readTable, which in turn is used in quite a lot of places and has no notion of query id... Also, MaterializedResult is compared using all its fields (or at least, it's intended to be - see #7204...), and we probably would not want the id to affect that comparison - so maybe it's better to leave it out of the MaterializedResult

BTW is there anything (except for the long no-rebase period) that stops us from merging #7204?